examples/quickstart-pennylane/quickstart_pennylane/client_app.py

```diff
@@ -0,0 +1,100 @@
+"""quickstart-pennylane: A Flower / Pennylane Quantum Federated Learning app."""
+
+import torch
+from flwr.app import ArrayRecord, Context, Message, MetricRecord, RecordDict
+from flwr.clientapp import ClientApp
+
+from quickstart_pennylane.task import QuantumNet, load_data
+from quickstart_pennylane.task import test as test_fn
+from quickstart_pennylane.task import train as train_fn
+
+# Flower ClientApp
+app = ClientApp()
+
+
+@app.train()
+def train(msg: Message, context: Context) -> Message:
+    """Train the quantum neural network on local data."""
+    
+    # Read quantum parameters from configuration
+    n_qubits = context.run_config.get("n-qubits", 4)
+    n_layers = context.run_config.get("n-layers", 3)
+    
+    # Load the model and initialize it with the received weights
+    model = QuantumNet(num_classes=10, n_qubits=n_qubits, n_layers=n_layers)
+    model.load_state_dict(msg.content["arrays"].to_torch_state_dict())
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    model.to(device)
+    
+    # Load the data
+    partition_id = context.node_config["partition-id"]
+    num_partitions = context.node_config["num-partitions"]
+    batch_size = context.run_config["batch-size"]
+    trainloader, valloader = load_data(partition_id, num_partitions, batch_size)
+    
+    print(f"Client {partition_id}/{num_partitions} starting training...")
+    print(f"Training data size: {len(trainloader.dataset)}")
+    print(f"Validation data size: {len(valloader.dataset)}")
+    
+    # Call the training function
+    results = train_fn(
+        model,
+        trainloader,
+        valloader,
+        context.run_config["local-epochs"],
+        msg.content["config"]["lr"],
+        device,
+    )
+    
+    print(f"Training completed. Train loss: {results['train_loss']:.4f}, "
+          f"Val accuracy: {results['val_accuracy']:.2f}%")
```

Comment:
```suggestion
```
same as above. Note that the aggregated `train_loss` and `val_accuracy` is shown at the end of each round.